### PR TITLE
Fix IS locking/crashes

### PR DESCRIPTION
**Pull request by UdjinM6**
Slightly tested, pls do more test (esp. on MN side)

### CODE REVIEW

File: src/instantx.cpp [hunk 1]
```diff
@@ -351,6 +355,7 @@ bool ProcessTxLockVote(CNode* pnode, CTxLockVote& vote)
 
 void ProcessOrphanTxLockVotes()
 {
+    LOCK2(cs_main, cs_instantsend);
```
Comment (tgflynn) [inline src/instantx.cpp]: Why lock cs_main here ?

**Comment by 1fabunicorn** [comment]
Hey there. We over on the dashnation are working on testing this using a spam script for instant send transactions

File: src/instantx.cpp (hunk 1 above)
Comment (UdjinM6) [inline src/instantx.cpp]: there is a deadlock otherwise

Comment (tgflynn) [inline src/instantx.cpp]: I guess because ProcessTxLockRequest is called from main.cpp with cs_main held.

**Comment by tgflynn** [comment]
utACK

**Comment by UdjinM6** [comment]
Let's see if @schinzelh 's nodes survive next few hours.

**Pull request merged**